processing/src/main/java/org/apache/druid/query/filter/BoundDimFilter.java

```diff
@@ -361,107 +361,107 @@ public String toString()
     class BoundLongPredicateSupplier implements Supplier<DruidLongPredicate>
```

Comment:
This class is actually a duplicate of `Suppliers.memoize()`. So code can be `return Suppliers.memoize(BoundDimFilter::createLongPredicate)`, assuming `updateSyncedDruidLongPredicate()` is renamed to `createLongPredicate()` and moved one level up.

Same below.

Reply:
1. Modified the code.
2. Have imported eclipse_formatting.xml to the workspace, but  style check still failed.

Reply:
Have imported eclipse_formatting.xml to the workspace, but style check still failed.

Reply:
In this case, it's better to turn off auto-formatting in your IDE and accurately revert unrelated changes.

For a longer-term solution, if you plan to contribution to the project regularly, it's better to consult developers at dev@druid.apache.org

Reply:
How about the change of the code logic to u?

Reply:
LGTM

Reply:
> In this case, it's better to turn off auto-formatting in your IDE and accurately revert unrelated changes.
> 
> For a longer-term solution, if you plan to contribution to the project regularly, it's better to consult developers at [dev@druid.apache.org](mailto:dev@druid.apache.org)

In the line 58 of eclipse_formatting.xml, it should be changed to:
<setting id="org.eclipse.jdt.core.formatter.insert_new_line_before_catch_in_try_statement" value="insert"/> instead of "do not insert".  By making this change, "catch" block will be alone on a line to in line with CI standard. Thus eclipse auto-formatting will be correct. I think the xml should be updated although I guess no one else than me is still using eclipse...

Reply:
> In this case, it's better to turn off auto-formatting in your IDE and accurately revert unrelated changes.
> 
> For a longer-term solution, if you plan to contribution to the project regularly, it's better to consult developers at [dev@druid.apache.org](mailto:dev@druid.apache.org)

In the line 58 of eclipse_formatting.xml, it should be changed to:
"<setting id="org.eclipse.jdt.core.formatter.insert_new_line_before_catch_in_try_statement" value="insert"/>" instead of "do not insert". By making this change, "catch" block will be alone on a line to in line with CI standard. Thus eclipse auto-formatting will be correct. I think the xml should be updated although I guess no one else than me is still using eclipse...

Reply:
In the line 58 of eclipse_formatting.xml, it should be changed to:
<setting id="org.eclipse.jdt.core.formatter.insert_new_line_before_catch_in_try_statement" value="insert"/> instead of "do not insert". By making this change, "catch" block will be alone on a line to in line with CI standard. Thus eclipse auto-formatting will be correct. I think the xml should be updated although I guess no one else than me is still using eclipse...